src/docfx/lib/JsonUtility.cs

```diff
@@ -181,12 +203,75 @@ private static void Traverse(this JToken token, List<Error> errors, List<JToken>
                     }
                     else
                     {
-                        prop.Value.Traverse(errors, nullNodes, prop.Name);
+                        prop.Value.TraverseForNullValueValidation(errors, nullNodes, prop.Name);
                     }
                 }
             }
         }
 
+        private static void TraverseForMismatchingTypeValidation(this JToken token, List<Error> errors, Type type)
+        {
+            if (token is JArray array)
+            {
+                foreach (var item in token.Children())
+                {
+                    item.TraverseForMismatchingTypeValidation(errors, type);
```

Comment:
>type [](start = 70, length = 4)

this looks incorrect if the type is an array of array of object:

A[][]

Reply:
Will add a test for it

---
In reply to: [205662738](https://github.com/dotnet/docfx/pull/3112#discussion_r205662738) [](ancestors = 205662738)